src/framework/components/script-esm/component.js

```diff
@@ -0,0 +1,327 @@
+     * cloned entities. This method remaps the script attributes for this entity and it assumes
+     * that this entity is the result of the clone operation.
+     *
+     * @param {ScriptESMComponent} oldScriptComponent - The source script component that belongs to
+     * the entity that was being cloned.
+     * @param {object} duplicatedIdsMap - A dictionary with guid-entity values that contains the
+     * entities that were cloned.
+     * @private
+     */
+    resolveDuplicatedEntityReferenceProperties(oldScriptComponent, duplicatedIdsMap) {
+
+        // TODO - run over old script component, any entities found, re-point them
+    }
+
+    /**
+     * Detect if script is attached to an entity.
+     *
+     * @param {string} moduleSpecifier - The module specifier to search for
+     * @returns {boolean} If script is attached to an entity.
+     * @example
+     * if (entity.module.has('path/to/module.mjs')) {
+     *     // entity has script
+     * }
+     */
+    has(moduleSpecifier) {
+        return this.modules.has(moduleSpecifier);
+    }
+
+    /**
+     * Get a script instance (if attached).
+     *
+     * @param {string} moduleSpecifier - The
+     * name or type of {@link ScriptType}.
+     * @returns {import('../../script/script-type.js').ScriptType|null} If script is attached, the
+     * instance is returned. Otherwise null is returned.
+     * @example
+     * const controller = entity.module.get('module');
+     */
+    get(moduleSpecifier) {
+        return this.modules.get(moduleSpecifier);
+    }
+
+    /**
+     * Create a script instance and attach to an entity script component.
+     *
+     * @param {string} moduleSpecifier - The module specifier used to import the ES module.
+     * @param {object} [args] - Object with arguments for a script.
+     * @param {boolean} [args.enabled] - If script instance is enabled after creation. Defaults to
+     * true.
+     * @param {object} [args.attributes] - Object with values for attributes (if any), where key is
+     * name of an attribute.
+     * @param {boolean} [args.preloading] - If script instance is created during preload. If true,
+     * script and attributes must be initialized manually. Defaults to false.
+     * @param {number} [args.ind] - The index where to insert the script instance at. Defaults to
+     * -1, which means append it at the end.
+     * @returns {*} Returns an instance of a
+     * ES Module if successfully attached to an entity, or null if the import fails
+     * @example
+     * entity.module.create('moduleSpecifier', {
+     *     attributes: {
+     *         speed: 4
+     *     }
+     * });
+     */
+    create(moduleSpecifier, args) {
+
+        const { attributes: definedAttributes } = args;
+
+        let finalUrl = moduleSpecifier;
+
+        // #if _ASSET_BASE_URL
+        // eslint-disable-next-line no-undef
+        finalUrl = $_ASSET_BASE_URL + this.system.app.assets.prefix +  moduleSpecifier;
+        // #endif
```

Comment:
![image](https://github.com/playcanvas/engine/assets/5236548/3a874cdd-335c-4510-bdf8-976ea3b55ebb)

Now the code is "rollup dependent", the reference error is caused by trying to use it via:

```js
import * as pc from '/playcanvas-engine-mark/src/index.js';
window.pc = pc;
```

I don't really like the entire idea of making this a build-time-only-thingy in the first place, it's just causing problems like this + extra build complications. IMO everything in JS should be easily configurable at runtime, if it's worth to configure it.

Reply:
Ah that's my bad. It should be the following, which should avoid that error.

```javascript
/* #if _ASSET_BASE_URL
const finalUrl = $_ASSET_BASE_URL + this.system.app.assets.prefix +  moduleSpecifier;
// #else */
const finalUrl = moduleSpecifier;
// #endif
```

Your point is valid tho, we shouldn't rely on specific bundle options, but maybe in this scenario it's ok, since it's only for local development (use_local_engine), and by default it remains just a comment. I guess we do the same for the debug flag